cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -281,15 +282,18 @@ public void onBecomeLeaderFromFollower(Message message, NotificationContext cont
           }
         }
 
+        // At this point, this replica has caught up to have the highest sequence number, except the replicas
+        // it failed to contact with.
         LOG.error("local.seq:" + localSeq + " replica:[" + numActiveReplicas + "/"+ totalReplicas + "]");
+
         // 3-Node failure case
         if (localSeq <= 0) {
           PartitionState lastState = stateManager.getState();
           if (!lastState.isValid()) {
             LOG.error("Invalid last state [part: " + partitionName + "] State:" + lastState.toString());
           } else {
             double tolerance = 0.5; 
-            if (localSeq >= (tolerance * lastState.seqNum)) {
+            if (localSeq >= (tolerance * lastState.seqNum)) { // FIXME(KV-5087): it seems this is never true since localSeq <= 0
```

Comment:
@premkumr can you confirm this?

Reply:
Good Catch. Jia .. This check should be on the >0 side ..